pkg/prompts/prompts.go

```diff
@@ -127,6 +143,65 @@ func validateBiggerThanZero(input string) error {
 	return nil
 }
 
+func (r *realPrompter) CaptureListDecision(
+	prompter Prompter, // we need this in order to be able to run mock tests
+	prompt string,
+	capture func(prompt string, args any) (any, error),
+	capturePrompt string,
+	label string,
+	info string,
+	arg any,
+) ([]any, bool, error) {
```

Comment:
maybe... this is a great opportunity to use go generics, and so avoid the extra for loop after calling this function to convert types

Reply:
I am afraid I don't understand this suggestion. In any case, generics help where the _signature_ is the same but the _types_ are different. This is exactly not the case here. Also, it took me a lot of time to get to this design, and trying to change it to generics will take more time again, which I don't think is necessary.

Reply:
I don't believe it shall take much time. Here you have the same interface, and will call it with different captured types.